### PR TITLE
test(evals): expand eval_data Q&A coverage

### PR DESCRIPTION
**Pull request by ItzikEzra-rh**
Added 51 evaluation queries across 47 conversation groups. 
In five local runs, the suite achieved a 96–100% pass rate.

## Summary by CodeRabbit

* **Tests**
  * Added a large set of conversational test cases covering cluster creation follow-ups, operator handling (including virtualization), platform/architecture queries (including s390x), VIP/configuration scenarios, Day‑1/Day‑2 operations, cloud boundaries, safety/refusal edge cases, and many boundary tests.
  * Updated one refusal-case expectation to a shorter, broader phrasing.
* **Tests**
  * Added an automated verification script to query the service and determine the latest stable OpenShift release.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: test/evals/eval_data.yaml [hunk 1]
```diff
@@ -531,3 +531,399 @@
       eval_query: What is the name of the cluster that I want to create?
       eval_types: [response_eval:intent]
       expected_intent: test-cluster
+- conversation_group: add_invalid_operator
```
Comment (eranco74) [inline test/evals/eval_data.yaml]: Please add an empty line between converstaion_group(s)

File: test/evals/eval_data.yaml [hunk 2]
```diff
@@ -531,3 +531,399 @@
...
+  description: Test adding an unavailable operator to a cluster
+  conversation:
+  - eval_id: add_invalid_operator
+    eval_query: Add the 'openshift-storage' operator to cluster 12345678-1234-1234-1234-1234567890.
```
Comment (eranco74) [inline test/evals/eval_data.yaml]: This might fail due to "cluster doesn't exist"

File: test/evals/eval_data.yaml [hunk 3]
```diff
@@ -531,3 +531,399 @@
...
+  - eval_id: add_operator_virtualization
+    eval_query: I want to run VMs on my cluster. How do I enable that?
+    eval_types:
+    - response_eval:intent
+    expected_intent: Recommending the OpenShift Virtualization operator and offering to add it to a cluster if the cluster ID is provided.
+- conversation_group: ask_about_cpu_arch
+  description: Ask what CPU architectures are supported for cluster creation
+  conversation:
+  - eval_id: ask_about_cpu_arch
+    eval_query: What CPU architectures can I use for my cluster?
```
Comment (eranco74) [inline test/evals/eval_data.yaml]: We should add this query to another conversation and ask it after creating a cluster with the defaults (without mentioning the cpu arch during the create)

File: test/evals/eval_data.yaml [hunk 4]
```diff
@@ -531,3 +531,399 @@
...
+    - response_eval:intent
+    expected_intent: Listing the available CPU architectures supported by the Assisted Installer, such as x86_64, aarch64, ppc64le, etc.
+- conversation_group: ask_about_platform_none
+  description: User asks for clarification on the 'none' platform type
+  conversation:
+  - eval_id: ask_about_platform_none
+    eval_query: What does it mean when the platform is 'none'?
+    eval_types:
+    - response_eval:intent
+    expected_intent: Explaining that the 'none' platform type is used for installations that do not have integration with an underlying infrastructure provider like vSphere or Nutanix. It is required for Single Node OpenShift (SNO) and can be used for baremetal-like deployments without provider-specific features.
```
Comment (eranco74) [inline test/evals/eval_data.yaml]: It should also mention the differences between None platform and Baremetal platform

File: test/evals/eval_data.yaml [hunk 5]
```diff
@@ -531,3 +531,399 @@
...
+    - response_eval:intent
+    expected_intent: Declining the request and clarifying that it is an expert on Assisted Installer for on-premise deployments, not Red Hat OpenShift Service on AWS (ROSA).
+- conversation_group: ask_about_s390x_support
+  description: Ask about a specific CPU architecture
+  conversation:
+  - eval_id: ask_about_s390x_support
+    eval_query: Can I install OpenShift on an s390x machine?
+    eval_types:
+    - response_eval:intent
+    expected_intent: Confirming that 's390x' is a supported CPU architecture and offering to help create a cluster with that configuration.
```
Comment (eranco74) [inline test/evals/eval_data.yaml:586]: Add a follow up query asking: list the avilable openshift versions for my env
validation should check that the version list fits the CPU arch

File: test/evals/eval_data.yaml [hunk 6]
```diff
@@ -531,3 +531,399 @@
...
+- conversation_group: ask_for_latest_version
+  description: Ask for the latest stable OpenShift version available for installation
+  conversation:
+  - eval_id: ask_for_latest_version
+    eval_query: What's the latest stable OCP version I can install?
+    eval_types:
+    - response_eval:intent
+    expected_intent: Retrieve the list of available versions, identify the latest non-pre-release version, and provide it to the user.
```
Comment (eranco74) [inline test/evals/eval_data.yaml]: I think this require a script for validation

File: test/evals/eval_data.yaml (hunk 4 above)
Comment (ItzikEzra-rh) [reply test/evals/eval_data.yaml]: so do you mean something like this?


 eval_query: What does it mean when the platform is 'none'? What's the difference between none and baremetal platform?

 expected_intent: Explaining that the 'none' platform is exclusively for Single Node OpenShift (SNO) clusters, while 'baremetal' is for multi-node clusters on physical hardware. The key differences include SNO vs multi-node deployment, infrastructure management responsibility, and VIP requirements (baremetal requires VIPs while none/SNO does not).
